src/libespeak-ng/translate.c

```diff
@@ -1788,78 +1788,159 @@ static int EmbeddedCommand(unsigned int *source_index_out)
 	return 1;
 }
 
-// handle .replace rule in xx_rules file
-static int SubstituteChar(Translator *tr, unsigned int c, unsigned int next_in, int *insert, int *wordflags)
+static bool GetNextTranslateChars(int **from, int **to, bool next_step)
 {
-	int ix;
-	unsigned int word;
-	unsigned int new_c, c2, c_lower;
-	int upper_case = 0;
-	static bool ignore_next = false;
-	const unsigned int *replace_chars;
+	// Set pointers to next 'from' and 'to' part for replacement
+	// next_step indicates non-first replacement group
+	// Return true if got data, false, if reached end
+	int *idx = *from;
+	if (next_step) {                      // first 'from' is pointed already
+		int spaces = 0;
+		while (*idx != 0 && spaces < 2) { // Get next 'from' position skipping to part
+			idx++;
+			if (*idx == ' ')
+				spaces++;
+		}
+		while (*idx == ' ') // skip remaining spaces, if exists
+			idx++;
+		//idx++;
+	}
+	*from = idx;
+	if (**from == 0)
+		return false;
+
+	while (*idx != 0 && *idx != ' ') {   // get next 'to' position
+		idx++;
+	}
+	while (*idx == ' ') // skip remaining spaces, if exists
+		idx++;
+	//*to = ++idx;
+	*to = idx;
```

Comment:
This is similarly confusing to the `from` logic for the same reason (it obscures where/how `to` is being modified).

It would be better if `*to = *from` was used before the first while loop, then use `*to` instead of `idx`.